crates/proto/src/lib.rs

```diff
@@ -48,3 +49,19 @@ pub use crate::xfer::retry_dns_handle::RetryDnsHandle;
 pub use crate::xfer::secure_dns_handle::SecureDnsHandle;
 #[doc(hidden)]
 pub use crate::xfer::{BufDnsStreamHandle, BufStreamHandle, MessageStreamHandle};
+
+use futures::Future;
+use tokio::runtime::Runtime;
+
+/// Generic executor.
+pub trait Executor {
```

Comment:
A good question I was just asked, is why not just call the global block_on statically in the tests, rather than using a trait?

Reply:
global block_on? You mean tokio's runtime.block_on?  Because I wanna use fuchsia's executor to run the future. 
With the trait, I am able to run the same test cases with fuchsia's executor.

Reply:
Yes, I realized after asking this that we need this functionality.